### DISCUSSION TITLE
unable to setup commands

### DISCUSSION THREAD

**Discussion by jefrygiglegroder** [discordjs/discord.js#9811] [Q&A] [answered]
this is the logs i get when i launch my bot

```
Started refreshing 10 application (/) commands.
Invalid Form Body
application_id[NUMBER_TYPE_COERCE]: Value "undefined" is not snowflake.
[14:45:55] minecraft bot ready: commandbot
[14:45:55] discord bot ready: mcdisco332#7928
```

and this is the code

```js
const { REST, Routes } = require("discord.js");
const { token, clientID, serverID } = require("./config.json");
const fs = require("node:fs");
const path = require("node:path");
const bot = require("./client");

const commands = [];
const commandsPath = path.join(__dirname, "commands");
const commandFiles = fs.readdirSync(commandsPath).filter((file) => file.endsWith(".js"));

for (const file of commandFiles) {
  const command = require(`./commands/${file}`);
  commands.push(command.data.toJSON());
}

const rest = new REST({ version: "10" }).setToken(token);

(async () => {
  try {
    console.log(`Started refreshing ${commands.length} application (/) commands.`);

    const data = await rest.put(Routes.applicationCommands(clientID, serverID), { body: commands });

    console.log(`Successfully reloaded ${data.length} application (/) commands.`);
    process.exit(0);
  } catch (error) {
    console.error(error.message);
  }
})();
```

**Comment by sdanialraza** [comment, marked as answer]
Looks like your `clientID` is undefined and you want to use `applicationGuildCommands` for deploying commands in a guild, `applicationCommands` is for deploying commands globally. Check out [this guide](https://discordjs.guide/creating-your-bot/command-deployment.html#guild-commands)

**Reply by jefrygiglegroder** [reply]
but now i am just getting "Unknown Application" from my loggs
if i set the config.json to guild id i just get the same errors as before but if it is the serverid it just says unknown application

**Reply by sdanialraza** [reply]
Your client ID is invalid then, are you getting it from [Discord Developer Portal](<https://discord.com/developers/applications>) > General Information > Application ID?

**Comment by jefrygiglegroder** [comment]
> Your client ID is invalid then, are you getting it from [Discord Developer Portal](https://discord.com/developers/applications) > General Information > Application ID?

not the minecraft bot does not join and i get this 
PS C:\Users\mrlag\OneDrive\Desktop\mineflayer-discord-bot-master> node src/register.js
Started refreshing 10 application (/) commands.
[15:39:24] minecraft bot ready: OBNinjaa
Successfully reloaded 10 application (/) commands.
PS C:\Users\mrlag\OneDrive\Desktop\mineflayer-discord-bot-master>

**Reply by sdanialraza** [reply]
Please join the [discord.js Discord server](https://discord.gg/djs) instead and I or someone will help you there, make sure to share the relevant code and errors.

**Reply by jefrygiglegroder** [reply]
ok thanks for the help anyways
